x/programs/test/programs/simple_token/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+use wasmlanche_sdk::Context;
+use wasmlanche_sdk::{public, state_keys, types::Address, Program};
+
+const INITIAL_SUPPLY: i64 = 123456789;
+
+/// The program state keys.
+#[state_keys]
+pub enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
+    /// The name of the token. Key prefix 0x1.
+    Name,
+    /// The symbol of the token. Key prefix 0x2.
+    Symbol,
+    /// The balance of the token by address. Key prefix 0x3 + address.
+    Balance(Address),
+
+    Allowance([Address; 2]),
```

Comment:
```suggestion
    Allowance(Address, Address),
```
with the comment above about `PrefixedBytes`. 

We should be careful here... Using an enum for state-keys might not actually make sense; the size of `StateKeys::Name` is necessarily the same size as `StateKeys::Allowance` (the type has a size, not each variant). I'm not suggesting we change this now, but we should probably benchmark if it makes a difference.